### PR TITLE
Support different main branches in auto sheriff

### PR DESCRIPTION
**Pull request by mai93**
Support different possible names of main branches in getting main build result in auto sheriff.

### CODE REVIEW

File: buildkite/bazel_auto_sheriff.py [hunk 1]
```diff
@@ -101,13 +101,14 @@ def __init__(self, project, pipeline, downstream_result):
 
 
     def _get_main_build_result(self):
-        build_info_list = self.client.get_build_info_list([
-            ("branch", "master"),
+        builds_filter = [("branch[]", b) for b in bazelci.MAIN_BRANCHES]
```
Comment (meteorcloudy) [inline buildkite/bazel_auto_sheriff.py:104]: It is possible that multiple "MAIN BRANCHES" exist for one pipeline, right? (like both google and master branches exists for the old pipeline). The the build result returned here may not from the actual default branch we want.

Can we use buildkite API to actually find out the default branch of the pipeline? See https://buildkite.com/docs/apis/rest-api/pipelines#get-a-pipeline and you may need to add a `get_pipeline_info` function near https://github.com/bazelbuild/continuous-integration/blob/6c94f47406bd20eeb43b5cfa0890cd29b79e5457/buildkite/bazelci.py#L723

**Review by meteorcloudy — CHANGES_REQUESTED** [review]
Thanks for fixing this!

File: buildkite/bazelci.py [hunk 2]
```diff
@@ -720,6 +724,20 @@ def _open_url(self, url, params=[]):
         except urllib.error.HTTPError as ex:
             raise BuildkiteException("Failed to open {}: {} - {}".format(url, ex.code, ex.reason))
 
+    def get_pipeline_info(self):
+    """Get details for a pipeline given its organization slug
```
Comment (meteorcloudy) [inline buildkite/bazelci.py:728]: Oh, we need to fix the indentation here.

**Review by meteorcloudy — CHANGES_REQUESTED** [review]

File: buildkite/bazelci.py (hunk 2 above)
Comment (mai93) [reply buildkite/bazelci.py]: Fixed, sorry about that.

**Review by meteorcloudy — APPROVED** [review]

**Pull request merged**